### ISSUE TITLE
[BUG] avoid-unused-parameters reports parameters of abstract method as unused

### ISSUE THREAD

**Issue by gaetschwartz** [dart-code-checker/dart-code-metrics#436]
* **Dart code metrics version:** `4.2.0`
* **Dart sdk version:** `2.15.0-42.0.dev`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
include: package:flutter_lints/flutter.yaml

analyzer:
  plugins:
    - dart_code_metrics
  exclude:
    - "**/*.g.dart"
    - "**/*.freezed.dart"
  strong-mode:
    implicit-dynamic: false
    implicit-casts: false
  errors:
    included_file_warning: ignore
    invalid_annotation_target: ignore

linter:
  rules:
    avoid_print: false
    prefer_final_fields: true
    prefer_final_in_for_each: true
    prefer_final_locals: true
    prefer_const_constructors: true
    prefer_const_literals_to_create_immutables: true

dart_code_metrics:
  anti-patterns:
    - long-method
    - long-parameter-list
  metrics:
    cyclomatic-complexity: 20
    maximum-nesting-level: 5
    number-of-parameters: 6
    source-lines-of-code: 64
  metrics-exclude:
    - test/**
  rules:
    - no-boolean-literal-compare

    - avoid-returning-widgets:
        ignored-annotations:
          - allowReturningWidgets
    - prefer-extracting-callbacks
    - no-equal-then-else
    - binary-expression-operand-order
    # - avoid-non-null-assertion
    - avoid-unused-parameters
    # - no-magic-number:
    #     allowed: [2]
    # - no-object-declaration
    - avoid-unnecessary-setstate
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
/// A route from [from] to [to].
mixin RouteConnection {

  RouteConnection copyWithLegs(List<Leg> legs); // Dart Code Metrics is unhappy

}
```

**What did you expect to happen?**

Dart Code Metrics not reporting the the parameter of the abstract method of the mixin as unused.

**What actually happened?**

It did report the parameteras unused.

**Are you willing to submit a pull request to fix this bug?**

I'm not familiar with the project so I'm not sure.

**Comment by incendial** [comment]
Available in 4.2.1 🚀

**Comment by gaetschwartz** [comment]
@incendial Wow ! Thanks for the blazing fast fix !  🚀

**Comment by incendial** [comment]
@gaetschwartz you're welcome! If you have any feedback about the tool, we'd be happy to hear it.
